test/libyul/objectCompiler/nested_subobject_data_access.yul

```diff
@@ -0,0 +1,92 @@
+object "A" {
+    code {
+        sstore(0, datasize("B.C"))
+        sstore(1, datasize("B.C.dataC1"))
+        sstore(2, datasize("B.C.dataC2"))
+    }
+
+    data "dataA1" "abcdef"
+    data "dataA2" "123"
+
+    object "B" {
+        code {
+            sstore(0, datasize("B"))
+            sstore(1, datasize("dataB1"))
+            sstore(2, datasize("dataB2"))
+        }
+
+        data "dataB1" "abcdef"
+        data "dataB2" "123"
+
+        object "C" {
+            code {
+                sstore(0, datasize("dataC1"))
+            }
+
+            data "dataC1" hex"AA"
+            data "dataC2" "00"
+        }
+    }
+}
+// ----
+// Assembly:
+//     /* "source":42:57   */
+//   dataSize(sub_0.sub_0)
+//     /* "source":39:40   */
+//   0x00
+//     /* "source":32:58   */
+//   sstore
+//     /* "source":77:99   */
+//   dataSize(sub_0.sub_0)
+//     /* "source":74:75   */
+//   0x01
+//     /* "source":67:100   */
+//   sstore
+//     /* "source":119:141   */
+//   dataSize(sub_0.sub_0)
+//     /* "source":116:117   */
+//   0x02
+//     /* "source":109:142   */
+//   sstore
```

Comment:
This is wrong. It means that the sizes returned for `B.C.dataC1` and `B.C.dataC2` are the same and all equal to the size of `B.C` rather than being the sizes of the data elements.

This confirms what I said in https://github.com/ethereum/solidity/pull/13707/files#r1032515177, i.e. `Object::pathToSubObject()` should not be returning the path to the parent object. It should return the path to the data object instead.